### PR TITLE
fixes #10351 - Fixes issue with CVV restrictive promotion sequence in UI, BZ1169876

### PR DESCRIPTION
**Pull request by cfouant**
Depends upon https://github.com/Katello/bastion/pull/64

### CODE REVIEW

**Comment by ehelms** [comment]
Based on the following path:

Dev > Test > Stage

After promoting my content view version to Dev, then trying to skip Test to promote straight to Stage the modal that popped up told me the suggested environments which included "dev" even though it already exists there and I cannot promote to it via the UI.

**Comment by ehelms** [comment]
I feel like something should tell the users that green are suggested explicitly. Like a green square with a dash and then text saying suggested environments.

**Comment by cfouant** [comment]
@ehelms - can you test the promotion again? I cannot reproduce the bug you're mentioning.

**Comment by cfouant** [comment]
@ehelms - explicit message and star icon to indicate suggested environment. Please let me know if you are still getting that bug - I have not been able to reproduce.

**Comment by ehelms** [comment]
Providing a screen shot to display what I mentioned earlier.

envs

**Comment by cfouant** [comment]
@ehelms - okay, finally able to recreate and fix

**Comment by cfouant** [comment]
@ehelms - also tweaked the message for cases where single suggested environment, or no suggested environment.

**Comment by cfouant** [comment]
[test]

**Comment by ehelms** [comment]
[test]

**Comment by cfouant** [comment]
[test]

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/content-views/details/content-view-promotion.controller.js [hunk 1]
```diff
@@ -63,22 +67,52 @@ angular.module('Bastion.content-views').controller('ContentViewPromotionControll
...
+        $scope.suggestedEnvironmentMessage = function () {
+            var envs = _.uniq(_.pluck($scope.suggestedEnvironments, 'name'));
+
+            if (envs.length === 0) {
+                return "There are no environments suggested in the promotion path.";
+            } else if (envs.length === 1) {
+                return "Suggested environment is: ".concat(envs[0]);
+            } else {
+                return "Suggested environments are: ".concat(envs.join(', '));
+            }
```
Comment (ehelms) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/content-views/details/content-view-promotion.controller.js]: We often try to limit a function to a single return. So you could set the strings inside the if-else to a variable and then have the function return that variable at the end.

**Comment by cfouant** [comment]
@ehelms - YAY finally tests are passing

**Comment by ehelms** [comment]
ACK

**Pull request merged**